src/backend/gencomm.py

```diff
@@ -112,8 +112,13 @@ def populate_pupfns(pup_max_nesting, backend, blklens, builtin_types, builtin_ma
     yutils.display(OUTFILE, "yaksuri_%si_type_s *%s = (yaksuri_%si_type_s *) type->backend.%s.priv;\n" \
                    % (backend, backend, backend, backend))
     yutils.display(OUTFILE, "\n")
+    yutils.display(OUTFILE, "#ifdef KERNEL_NULL\n")
+    yutils.display(OUTFILE, "%s->pack = KERNEL_NULL;\n" % backend)
+    yutils.display(OUTFILE, "%s->unpack = KERNEL_NULL;\n" % backend)
+    yutils.display(OUTFILE, "#else\n")
```

Comment:
This seems messy.  `KERNEL_NULL` is a ze-specific macro, which is being checked for in `gencomm`, which is not ze-specific.  It might be better to replace this with one extra level of indirection.  For example, each driver can maintain an array of kernels (or pointers to kernels) and the pack/unpack variables can simply maintain indices within that array.  That way, we can use index `-1` to mean that this type does not have a pack/unpack kernel.

Reply:
I don't see why it is ze-specific macro. Any backend can define KERNEL_NULL. To clean it up, I can use KERNEL_NULL in this script for all backends. And define KERNEL_NULL to be "NULL" for cuda,   and "-1" for ze.

Reply:
It's defined in `yaksuri_zei.h`.  Ergo, it's ze-specific.  Yes, you can fix it in multiple ways.  But the current code is not one of those.  It needs to be changed.  Also, if you want to make it generic, we'll need a namespace to it: `YAKSURI

Reply:
Ok, I will fix it by naming it YAKSRI_KERNEL_NULL, and define it in each backend.